### PR TITLE
Home page

### PR DESCRIPTION
**Pull request by Simpcyclassy**

### CODE REVIEW

File: components/home/actions.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { UPDATE_REACTION, REQUEST_UPLOAD_LIST } from './actionTypes';
+
+/**
+ * Triggers request to react on a post
+ *
+ * @function
+ * @return {Object} The {@link actionTypes.REQUEST_UPLOAD_LIST REQUEST_UPLOAD_LIST}
+ * action.
+ */
+export function updateReaction(id, reaction) {
```
Comment (Way2nnadi) [inline components/home/actions.js]: Keep the pattern of this function the same as the one below

File: components/home/components/Home.jsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        <Icon type={type} style={ICON} onClick={onClick} />
+        {text}
+    </span>
+);
+function createReactionsIcon(item, updateReaction) {
+    const { like, dislike, maybe } = item.reactions;
+    const icons = [
+        { reaction: 'like', text: `${like}`, type: 'heart' },
+        { reaction: 'dislike', text: `${dislike}`, type: 'dislike' },
+        { reaction: 'maybe', text: `${maybe}`, type: 'meh' },
```
Comment (Way2nnadi) [inline components/home/components/Home.jsx]: what purpose does the key `reaction` serve here?

File: components/home/reducers.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+                ...state,
+                uploads: generateUploadsMap(),
+            };
+        }
+        case UPDATE_REACTION: {
+            const { uploads } = state;
+
+            return {
+                ...state,
+                uploads: updateUploadReaction(action.id, action.reaction, uploads),
```
Comment (Way2nnadi) [inline components/home/reducers.js]: destructure `action.payload` for `{ id, reaction }`

File: components/home/components/Home.jsx (hunk 2 above)
Comment (Simpcyclassy) [reply components/home/components/Home.jsx]: That's the unique `key` prop for the icons.

File: components/home/actionTypes.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+ * action creator.
+ *
+ * @type {String}
+ */
+export const UPDATE_REACTION = `${NAME}/UPDATE_REACTION`;
+
+/**
+ * Fired by the {@link actions.requestUploadList requestUploadList}
+ * action creator.
+ *
```
Comment (ctodmia) [inline components/home/actionTypes.js:14]: testing

File: components/home/actions.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import { UPDATE_REACTION, REQUEST_UPLOAD_LIST } from './actionTypes';
+
+/**
+ * Triggers request to react on a post
+ *
+ * @function
+ * @return {Object} The {@link actionTypes.REQUEST_UPLOAD_LIST REQUEST_UPLOAD_LIST}
+ * action.
+ */
+export const updateReaction = (id, reaction) => ({
```
Comment (enyeInc) [inline components/home/actions.js]: make this action follow the same pattern as `requestUploadList`

File: components/home/reducers.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    USER_UPLOADS.forEach(userUpload => {
+        const { id } = userUpload;
+
+        uploads.set(id, userUpload);
+    });
+
+    return uploads;
+}
+
+function updateUploadReaction(id, reaction, uploads) {
```
Comment (enyeInc) [inline components/home/reducers.js]: this should be `function updateUploadReaction(id, type, uploads) {...}`

**Comment by enyeInc** [comment]
@Simpcyclassy Your code is broken, fix the bug, then this should be good to go

File: components/home/reducers.js (hunk 6 above)
Comment (Simpcyclassy) [reply components/home/reducers.js]: Alright, that makes more sense

File: components/home/actions.js (hunk 5 above)
Comment (Simpcyclassy) [reply components/home/actions.js]: OK.

File: components/home/actionTypes.js (hunk 4 above)
Comment (Simpcyclassy) [reply components/home/actionTypes.js:14]: Good for testing now

File: components/home/components/Home.jsx (hunk 2 above)
Comment (Way2nnadi) [reply components/home/components/Home.jsx]: just use `type` as your `key`

**Comment by Way2nnadi** [comment]
@Simpcyclassy why did you merge this code, no one approved it

**Pull request merged**